src/Form/EntityEmbedDialog.php

```diff
@@ -159,16 +185,40 @@ public function buildSelectStep(array &$form, FormStateInterface $form_state) {
       }
     }
 
-    $form['attributes']['data-entity-id'] = array(
-      '#type' => 'entity_autocomplete',
-      '#target_type' => $entity_element['data-entity-type'],
-      '#title' => $label,
-      '#default_value' => $entity,
-      '#required' => TRUE,
-      '#description' => $this->t('Type label and pick the right one from suggestions. Note that the unique ID will be saved.'),
-    );
-    if ($bundles = $embed_button->getTypeSetting('bundles')) {
-      $form['attributes']['data-entity-id']['#selection_settings']['target_bundles'] = $bundles;
+    if (($entity_browser_id = $embed_button->getTypePlugin()->getConfigurationValue('entity_browser'))) {
+      /** @var \Drupal\entity_browser\EntityBrowserInterface $entity_browser */
+      $entity_browser = $this->entityTypeManager->getStorage('entity_browser')->load($entity_browser_id);
+      $this->entity_browser = $entity_browser->id();
+      $this->eventDispatcher->addListener(Events::REGISTER_JS_CALLBACKS, [$this, 'registerJSCallback']);
+
+      $form['attributes']['entity_browser']['#theme_wrappers'] = ['container'];
+      $form['attributes']['entity_browser']['browser'] = $entity_browser->getDisplay()->displayEntityBrowser();
+      $form['attributes']['entity_browser']['entity-id'] = [
+        '#type' => 'hidden',
+        '#default_value' => $entity ? $entity->id() : '',
+        '#attributes' => ['class' => ['eb-target']]
+      ];
+      $form['#attached']['library'][] = 'entity_browser/common';
+      $form['#attached']['drupalSettings']['entity_browser'] = [
+        'field_settings' => [$entity_browser->getDisplay()->getUuid() => ['cardinality' => 1]]
```

Comment:
This configuration value was moved to a different part of the settings array. See https://github.com/drupal-media/entity_browser/commit/e712253121cb3fef3abf1909da1e2bfd3bc35938